framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -1033,18 +1019,30 @@ protected void processConditionalPostconditions(
                                 flowExprContext,
                                 analysis.atypeFactory.getPath(tree),
                                 false);
-                if (result) {
-                    thenStore.insertValue(r, anno);
+                if (p instanceof ConditionalPostcondition) {
+                    if (((ConditionalPostcondition) p).annoResult) {
+                        thenStore.insertValue(r, anno);
+                    } else {
+                        elseStore.insertValue(r, anno);
+                    }
                 } else {
-                    elseStore.insertValue(r, anno);
+                    thenStore.insertValue(r, anno);
                 }
             } catch (FlowExpressionParseException e) {
+                Result r = e.getResult();
```

Comment:
Would it make sense to more nicely encapsulate this within the `FlowExpressionParseException`? At the moment there are two kinds of parse exceptions and we could either have separate subclasses or one subclass with an enum kind. Or simply add a helper method in that class to encapsulate this logic?
Inspecting the result and comparing the message key seems a brittle.